libs/ml/include/ml/dataloaders/dataloader.hpp

```diff
@@ -50,13 +50,19 @@ class DataLoader
 
   virtual ~DataLoader() = default;
 
-  virtual ReturnType GetNext() = 0;
+  virtual ReturnType GetNext(bool is_validation = false) = 0;
```

Comment:
rather than pass a bool into every function, it seems more logical to have a mode flag for the dataloader, i.e.:

enum class DataLoaderMode
{
    TRAIN,
    VALIDATE,
    TEST,
    INVALID
} 

class DataLoader
{

private:
  DataLoaderMode mode_ = DataLoaderMode::INVALID;
}